### PR TITLE
[proof-new] Adding support for corner case of transitivity simulating MERGED_THROUGH_CONSTANTS

### PR DESCRIPTION
**Pull request by HanielB**

### CODE REVIEW

File: src/theory/uf/eq_proof.h [hunk 1]
```diff
@@ -182,14 +182,44 @@ class EqProof
...
       std::vector<Node>& premises,
       CDProof* p,
       std::unordered_set<Node, NodeHashFunction>& assumptions) const;
 
+  /** Expand coarse-grained transitivity steps for theory disequalities
+   *
+   * Currently the equality engine can represent disequality reasoning of theory
+   * symbols in a rather coarse-grained manner with EqProof. This is the case
+   * when EqProof is
+   *            (= t1 c1) (= t2 c2)
```
Comment (ajreynol) [inline src/theory/uf/eq_proof.h:198]: Missing paren.

File: src/theory/uf/eq_proof.h [hunk 2]
```diff
@@ -182,14 +182,44 @@ class EqProof
...
+   *  --------------------------------------------------------- TR
+   *           (= (= t1 t2) false)
+   *
+   * @param conclusion the conclusion of the (possibly) coarse-grained
+   * transitivity step
+   * @param premises the premises of the (possibly) coarse-grained
+   * transitivity step
+   * @param p a pointer to a CDProof to store the conversion of this EqProof
+   * @return True if the EqProof transitivity step is the above case,
+   * symbolizing that the ProofNode justifying the conclusion has already been
```
Comment (ajreynol) [inline src/theory/uf/eq_proof.h:216]:
```suggestion
   * indicating that the ProofNode justifying the conclusion has already been
```

File: src/theory/uf/eq_proof.h (hunk 1 above)
Comment (HanielB) [reply src/theory/uf/eq_proof.h:198]: I'm not sure I understand.

Comment (ajreynol) [reply src/theory/uf/eq_proof.h:198]: Oh, I was confused I thought it was equality between equalities, guess not.

**Pull request merged**